### PR TITLE
feat: disable buttons on load or no items and show progress circular in chip while loading

### PR DESCRIPTION
**Pull request by ferferga**
That way we avoid popping or bad inputs while content is not loaded.

https://user-images.githubusercontent.com/10274099/105354474-e6f49780-5bf0-11eb-9010-aa4546093b84.mp4

### CODE REVIEW

**Comment by ThibaultNocchi** [comment]
IMO what I'd do is set the disabled props to not required and false by default. What do you think?

File: pages/library/_viewId.vue [hunk 1]
```diff
@@ -7,10 +7,14 @@
       <v-chip v-if="!loading" small class="ma-2 hidden-sm-and-down">
         {{ itemsCount }}
       </v-chip>
+      <v-chip v-else class="ma-2 hidden-sm-and-down">
+        <v-progress-circular width="2" indeterminate size="15" />
+      </v-chip>
```
Comment (ThibaultNocchi) [inline pages/library/_viewId.vue:12]:
```suggestion
      <v-chip :small="!loading" class="ma-2 hidden-sm-and-down">
        <template v-if="!loading">{{ itemsCount }}</template>
        <v-progress-circular v-else width="2" indeterminate size="15" />
      </v-chip>
```

**Review by ThibaultNocchi — CHANGES_REQUESTED** [review]

**Comment by ferferga** [comment]
@ThibaultNocchi Good point, didn't think about that ^^

I don't mind. However, we're not using those components anywhere else, so this might be a bit moot. And, if at some point we do, having them required by default makes it easy to "force" us to remember that we need to set that prop for better UX, as console messages will pop everywhere reminding that.

But, as said, don't mind it. Thoughts @MrTimscampi @Camc314 ?

**Comment by MrTimscampi** [comment]
> IMO what I'd do is set the disabled props to not required and false by default. What do you think?

Agreed, it's simply passing something to the component below it, so it should use the same thing, which is false by default and not required.

File: pages/library/_viewId.vue [hunk 2]
```diff
@@ -4,13 +4,15 @@
       <span class="text-h6 hidden-sm-and-down">
         {{ collectionInfo.Name }}
       </span>
-      <v-chip v-if="!loading" small class="ma-2 hidden-sm-and-down">
-        {{ itemsCount }}
+      <v-chip :small="!loading" class="ma-2 hidden-sm-and-down">
+        <template v-if="!loading">{{ itemsCount }}</template>
+        <v-progress-circular v-else width="2" indeterminate size="15" />
```
[the remark this answers is not in the document]
Comment (ferferga) [reply pages/library/_viewId.vue:9]: Good point 👍.

**Review by ThibaultNocchi — APPROVED** [review]

**Review by MrTimscampi — APPROVED** [review]

**Pull request merged**